en/codegen/layout-codegen.md

```diff
@@ -12,56 +12,120 @@ For each selected Artboard or Group, a single Angular Component is created. Each
 
 ## Layouts
 
-Layouts are rendered with absolute position and sizes. Each component has its own position (left, top) and size (width, height) that determine how it will be rendered on the page.
+Sketch arranges artboards of projects through absolute positioning, using properties such as Top, Left, Width and Height. Often these designs have to represent fluid and responsive applications. Most web applications built from such designs use modern paradigms such as flex and grid display containers. The job of the Code generation is to get as close as possible to the application, a developer will build from a given design. This requires the Code generation to go through a set of heuristics and standing on them to produce fluid design from the absolute/static one in Sketch.
+
+Code generation reads Sketch files and uses the layout properties defined by native Sketch schema. Please be aware, if you use a 3rd party Sketch plugin to create and maintain the Sketch drawing layout, they typically have their own set of properties that code generation is not aware of and will not use when creating the HTML and CSS.
+Code generation respects and uses Sketch groups when creating the layout. Code generation renders HTML CSS that uses [Flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/). Flexbox lays out elements in either a column or row.
+Sketch container elements such as a group or an artboard are rendered as divs with flex CSS applied to them.
+
+### How it works
+Main idea of building responsive layout is to combine the components into groups. This creates a structure, which the developer expects. From Code generation point of view, and because of the Flexbox, these groups can contain either rows or columns. After the layout is created, components are placed in a hierarchy of rows and columns depending on their positions relative to other components and their respective Sketch groups.
+
+In short, the algorithm starts by creating a row for the first component and stores in it the components, considered part of that row (laying on the same X-axis as the row). Once it exhausts the components in the artboard and creates all the rows for them, it uses the same principle, but for columns. For each of these rows creates columns, the components become part of the same column if they are on the same Y-axis. It repeats rows and columns until there is no need to create more groups.
+
+
+<img class="responsive-img" src="../images/layout_codegen_structure.png" />
+
+The fluid layout is in a tree structure. The branches and the root are rows and columns and the leaves are components.
+
+## Heuristics and rules
+In order to be properly adjust the elements in HTML, the Code generation applies heuristics and rules to get responsive design from statically positioned elements.
```

Comment:
In order to be properly adjust the elements -> In order to properly position the elements in HTML